### PR TITLE
Implement auth code grant.

### PR DESCRIPTION
**Pull request by DFurnes**
#### Changes

This pull request implements the authentication code grant in Northstar PHP! This is done using PSR-7 requests/responses so that the package continues to be framework-agnostic for Drupal and our frameworkless Quicksilver friends. 😄

I've also renamed the `OAuthRepository` to `OAuthBridge` since it's not _really_ a repository at this point (it's just a generic way for the library to talk to any type of host application) and so implying it just abstracted storage seemed wrong. 🌉

References DoSomething/TeamRocket#27.

---

For review: @angaither @weerd

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,36 +110,71 @@ class Inspire
...
+    /**
+     * Handle a login request to the application.
+     *
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function getLogin(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        return Northstar::authorize($request, $response, $this->redirectTo);
```
Comment (weerd) [inline README.md]: You probs need to indicate to `use DoSomething\Northstar\Northstar;` right?

**Review by weerd — APPROVED** [review]
Just a quick comment on the docs. Looks good otherwise!

File: README.md (hunk 1 above)
Comment (DFurnes) [reply README.md]: Agh yeah, it's using the facade there (I know, I know 😆) but it's confusing. I'd done it so the docs didn't have all the extra boilerplate of injecting the class. I replaced it with `app('northstar')` in 678c554.

Comment (DFurnes) [reply README.md]: Maybe we should include a `northstar()` (or I guess `gateway()`!!) helper method in the library in the future for something like that.

**Pull request merged**